### PR TITLE
fix ansible lint action

### LINKED ISSUE #30: Fix Ansible-Lint Action
ansible-lint github action fails every day. We need to fix this.

### PR DESCRIPTION
**Pull request by beechesII**
- changed linting task completely
- moved .ansible-lint to .config/ansible-lint.yml
- changed version for ansible-lint from main to v6
- fixed most of linting issues

Closes #30

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -47,20 +47,16 @@ jobs:
           path: ansible_collections/t_systems_mms/ansible_collection_icinga
 
       - name: Run Linting - icinga_agent
-        uses: ansible/ansible-lint-action@master
+        uses: ansible/ansible-lint-action@main
```
[the remark this answers is not in the document]
Comment (rndmh3ro) [reply .github/workflows/CI.yml]: renovate then updates the versions. :)

[the remark this answers is not in the document]
Comment (schurzi) [reply .github/workflows/CI.yml]: not only that. A year ago there was a global issue and I remember a very specific advice to only use the version tags.

[the remark this answers is not in the document]
Comment (schurzi) [reply .github/workflows/CI.yml]: what I remember is this: https://github.blog/changelog/2021-01-21-github-actions-short-sha-deprecation/
with a link to: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

So the general advice is to use the full commit SHA. Version Tags should only be used on trusted Actions.

[the remark this answers is not in the document]
Comment (beechesII) [reply .github/workflows/CI.yml]: I tagged the ansible/ansible-lint-action to v6, but we can also use full commit SHA. Please give me a hint what you like to have tag or full commit SHA.

[the remark this answers is not in the document]
Comment (schurzi) [reply .github/workflows/CI.yml]: I say using a version tag ist ok in this case, since we generally trust the Ansible project.

**Comment by xFuture603** [comment]
Changes tested on one of our nodes. LGTM!

**Review by xFuture603 — APPROVED** [review]

File: roles/icinga_agent/tasks/main.yml [hunk 2]
```diff
@@ -86,29 +88,38 @@
...
 
     - name: generate ticket and save it as a variable
-      ansible.builtin.shell: /usr/sbin/icinga2 pki ticket --cn {{ ansible_hostname }} --salt {{ icinga_agent_salt }}
+      ansible.builtin.command: >
+        /usr/sbin/icinga2 pki ticket --cn {{ ansible_hostname }} --salt {{ icinga_agent_salt }}
       environment:
         LD_LIBRARY_PATH: "/usr/lib64"
       register: ticket
+      changed_when: ticket.rc != 0
+      failed_when: ticket.rc != 0
```
Comment (schurzi) [inline roles/icinga_agent/tasks/main.yml:97]: the `changed_when` and `failed_when` conditions are the same, this looks wrong to me. Can you explain?

Comment (beechesII) [reply roles/icinga_agent/tasks/main.yml:97]: This is to pass the ansible-lint check no-changed-when: `Commands should not change things if nothing needs doing`. 

If the Returncode of the command is 0 than the task is `ok` and not `changed`. Is the Returncode of the command is 0 then the task is not failed. 

```
TASK [t_systems_mms.ansible_collection_icinga.icinga_agent : generate ticket and save it as a variable] *********************************************************************************************************
task path: /home/foo/my-ansible/collections/ansible_collections/t_systems_mms/ansible_collection_icinga/roles/icinga_agent/tasks/main.yml:90
ok: [my-example-host] => changed=false
  cmd:
  - /usr/sbin/icinga2
  - pki
  - ticket
  - --cn
  - my-example-host
  - --salt
  - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  delta: '0:00:00.267765'
  end: '2022-08-17 09:44:58.978878'
  failed_when_result: false
  rc: 0
  start: '2022-08-17 09:44:58.711113'
  stderr: ''
  stderr_lines: <omitted>
  stdout: <my ticket> 
  stdout_lines: <omitted>
```

So both values are correct. 

See also: https://github.com/ansible/ansible-lint/issues/1681

Comment (schurzi) [reply roles/icinga_agent/tasks/main.yml:97]: I just checked with the Icinga2 repository: https://github.com/Icinga/icinga2/blob/ee705bb110e802f8cafd21bab2d8697b0a538b0a/lib/cli/pkiticketcommand.cpp#L37-L55

It seems they only generate a different exit code, if there was an error. My concern is, that using `changed_when`suggests, that this task could display a changed status in execution. I think in this case I would prefer to use `changed_when: false` to signify, that this task will never display a changed status.

Comment (rndmh3ro) [reply roles/icinga_agent/tasks/main.yml:97]: Or disable the specific ansible-lint rule for this one task.

Comment (beechesII) [reply roles/icinga_agent/tasks/main.yml]: I changed it to false: https://github.com/T-Systems-MMS/ansible-collection-icinga/pull/31/commits/2b7a467eec09a269cb2b41d5cfba2897bfb46e6a

**Pull request merged**